src/Recommendations/RuleModelRepository.php

```diff
@@ -147,10 +198,9 @@ public function upsert(RuleModel $ruleModel): RuleModel
         if ($ruleModel->isFromDatabase()) {
             return $ruleModel;
         }
-        $dataModel = $this->get($ruleModel);
-        if ($dataModel) {
-            $ruleModel->setRuleId($dataModel['rule_id']);
```

Comment:
When fetching a correction, `$dataModel['type']` was correction, but `$ruleModel->getType()` was either `article` or `research-article` depending on the URL request